packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -966,35 +1010,78 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
 
   final Map<int, Widget> _childWidgets = HashMap<int, Widget>();
   final SplayTreeMap<int, Element> _childElements = SplayTreeMap<int, Element>();
-  RenderBox _currentBeforeChild;
 
   @override
   void performRebuild() {
     _childWidgets.clear(); // Reset the cache, as described above.
     super.performRebuild();
-    _currentBeforeChild = null;
     assert(_currentlyUpdatingChildIndex == null);
     try {
+      final Set<int> affectedIndexes = Set<int>.from(_childElements.keys);
+      final Map<int, Element> newKeepAliveChildren = <int, Element>{};
+
+      int getNewIndex(Widget w){
+        if (w.key != null){
+          return widget.delegate.findIndexByKey(w.key);
+        } else {
+          return -1;
+        }
+      }
+
+      void removeChildIfNotRemapped(int index){
+        final Element child = _childElements[index];
+        final SliverMultiBoxAdaptorParentData originalParentData = child?.renderObject?.parentData;
+        if (originalParentData != null){
+          if (getNewIndex(child.widget) == -1) {
```

Comment:
Don't we already know at this point that the child will not be remapped? If it is getting remapped, it's removed from _childElements in line 1066?

Reply:
removed